### PR TITLE
CI: Install word list package for refguide-check

### LINKED ISSUE #13111: BUG, DOC: refguide check is failing
Likely because of gh-13109, an issue with refguide-check has snuck in to CI; see: https://github.com/scipy/scipy/pull/13110, where the refguide-check was reactivated in Azure and full log will be available for 10 days (I can reproduce locally on latest `master` as well)

A sample of the refguide failures:

```

File "doc/source/tutorial/csgraph.rst", line 47, in /home/vsts/work/1/s/tools/../doc/source/tutorial/csgraph.rst
Failed example:
    word_list = [word for word in word_list if len(word) == 3]
Exception raised:
    Traceback (most recent call last):
      File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest /home/vsts/work/1/s/tools/../doc/source/tutorial/csgraph.rst[2]>", line 1, in <module>
        word_list = [word for word in word_list if len(word) == 3]
    NameError: name 'word_list' is not defined

```

### PR DESCRIPTION
**Pull request by peterbell10**
#### Reference issue
Should fix gh-13111

### CODE REVIEW

**Comment by tylerjereddy** [comment]
Thanks, we may eventually want to have that dependency requirement spelled out right in the error message rather than looking for a missing file...

**Comment by peterbell10** [comment]
The tutorial text already describes the dependency:
>The system word lists consist of a file with one word per line. The following should be modified to use the particular word list you have available:

I don't think it's worth cluttering up the tutorial with error handling.

**Comment by peterbell10** [comment]
Refguide was still failing:

https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=9163&view=logs&j=4c925117-6155-59c4-749e-aa25fa289e87&t=2ed93502-b530-5c8d-5362-c637a4a1ad25&l=327

The script has `failOnStderr: true` but for some reason `refguide_check.py` writes the dots output to `stderr`:
https://github.com/scipy/scipy/blob/9d3bcca4485b648febacfb4570ea16b11c9de031/tools/refguide_check.py#L877-L881

**Comment by tylerjereddy** [comment]
> I don't think it's worth cluttering up the tutorial with error handling.

I disagree, the dependency requirement should have nothing to do with the tutorial proper and everything to with refguide_check.py specifying that something is needed before running.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -50,6 +50,9 @@ jobs:
       wheel
     displayName: 'Install deps'
     failOnStderr: true
+  - script: |
+      sudo apt-get install -y wamerican-small
+    displayName: 'Install word list (for csgraph tutorial)'
```
Comment (tylerjereddy) [inline azure-pipelines.yml:55]: Is this package easily available on all platforms and is there a good reason we can't add shims to refguide_check.py to indicate it is not present and either raise an error or skip checking that tutorial?

**Comment by tylerjereddy** [comment]
Merging the CI fix, thanks!

I think there's room for a small shim in the refguide machinery eventually though

File: azure-pipelines.yml (hunk 1 above)
Comment (peterbell10) [reply azure-pipelines.yml:55]: I can override `DocTestRunner.report_unexpected_exception` to handle `FileNotFoundError` differently. But I'm not sure what you'd want to do instead. The error is 

If your problem is just that the error isn't the last in the list, I've found a small shim to stop running the doctest on the first exception. I can even examine the exception to only do this for specific exceptions.

Comment (tylerjereddy) [reply azure-pipelines.yml:55]: I think I'm suggesting the doctests shouldn't run at all (don't even "collect" them) if a required dependency is missing, but maybe not worth the energy to pursue at this point.

Alternatively, a selective exclusion of that tutorial when the dep is missing. A bit like the argument for pytest xfail/skip in selective scenarios--sure, the error could be obvious but what is it really contributing to the test suite vs. a controlled handling when you know what is wrong.

**Pull request merged**